client.go

```diff
@@ -285,20 +301,30 @@ func (c *HTTPClient) getHTTPData(method string, path string, body url.Values, da
 	if err != nil {
 		return err
 	}
-	w, err := cli.Do(r)
-	if err != nil {
-		return err
-	}
-	resp := &Response{}
-	resp.Data = data
-	decoder := json.NewDecoder(w.Body)
-	err = decoder.Decode(resp)
-	if err != nil {
-		return err
-	}
-	if resp.Status != "ok" {
-		return fmt.Errorf(resp.Message)
+
+	// Contains retry logic if we decode a 500 error.
+	for i := 1; i <= maxRetryAttempts; i++ {
+		w, err := cli.Do(r)
+		if err != nil {
+			return err
+		}
+		resp := &Response{}
+		resp.Data = data
+		decoder := json.NewDecoder(w.Body)
+		err = decoder.Decode(resp)
+		if err != nil {
+			return err
+		}
+		if resp.Status != "ok" {
+			if (resp.Message != "Internal Server Error") || (i == maxRetryAttempts) {
```

Comment:
It'd be better to use resp.Code than the message.

Reply:
I thought i noticed that there were more frequent cases of `resp.Code == InternalServerErrorCode` rather than `resp.Message == "Internal Server Error", meaning we were retrying more often than we should. I might be wrong, but updated and tested